iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/Coordinator.java

```diff
@@ -625,8 +654,20 @@ public void cleanupQueryExecution(
       try (SetThreadName threadName = new SetThreadName(queryExecution.getQueryId())) {
         LOGGER.debug("[CleanUpQuery]]");
         queryExecution.stopAndCleanup(t);
+        boolean isUserQuery = queryExecution.isQuery() && queryExecution.isUserQuery();
+        if (isUserQuery) {
+          recordCurrentQueries(
+              queryExecution.getQueryId(),
+              queryExecution.getStartExecutionTime(),
+              queryExecution.getStartExecutionTime()
+                  + queryExecution.getTotalExecutionTime() / 1_000_000L,
+              queryExecution.getTotalExecutionTime(),
+              queryExecution.getExecuteSQL().orElse("UNKNOWN"),
```

Comment:
better to use `ContentOfQuerySupplier` which is the same as slow query recorder.